docs/developer/crypto.md

```diff
@@ -7,3 +7,83 @@ During encryption, each value in the original Secret is symmetrically encrypted
+Even if QRNG (Quantum Random Number Generator) are considered better than PRNG (Pseudo Random Number Generator) in a quantum cryptography context as well as in a non-quantum context, QRNG relies on a quantum mechanical phenomenon. It requires a physical device, therefore QRNG usage is out of Sealed Secrets scope, which will stay on the `crypto/rand` usage.
+
+### **Associated documentation**
+
+[Combining a quantum random number generator and quantum-resistant algorithms into the GnuGPG open-source software](https://doi.org/10.1515/aot-2020-0021)
+
+## **AES-256-GCM**
+
+### **Analysis**
+
+AES-256-GCM is quantum resistant.
+Grover algorithm can reduce the bruteforce of the key from 2²⁵⁶ to 2¹²⁸ which is still considered very secure.
+Nevertheless, since AES uses unchangeable 128 bits blocks, Grover algorithm can in some cases decrease the complexity of the bruteforce to 2⁶⁴.
+
+### **Recommendations**
+
+AES-256-GCM quantum security is not a concern.
+Cases with a bruteforce complexity of 2⁶⁴ are unlikely for Sealed Secret considering how AES is used in the project.
+Even assuming that 2⁶⁴ bruteforce is likely, it can still be considered secure today (but not in the long run).
+A recommendation is to look for a AES replacement that provide 128 bits post-quantum cryptographic security in any cases, such as ChaCha20-Poly1305. Applying this recommendation is considered low priority.
+
+### **Associated documentation**
+
+[Quantum Security Analysis of AES](https://eprint.iacr.org/2019/272.pdf)
+
+[Critics on AES-256-GCM](https://soatok.blog/2020/05/13/why-aes-gcm-sucks/)
+
+[Security Analysis of ChaCha20-Poly1305 AEAD](https://www.cryptrec.go.jp/exreport/cryptrec-ex-2601-2016.pdf)
+
+
+## **SHA-256**
+
+### **Analysis**
+
+SHA-256 is quantum resistant.
+Grover Algorithm can reduce the bruteforce from 2²⁵⁶ to 2¹²⁸ which is considered very secure.
+It is computationally cheaper to use a non-quantum algorithm to generate a collision than to employ a quantum computer.
+
+### **Recommendations**
+
+No recommendations about SHA-256.
+
+### **Associated documentation**
+[Cost analysis of hash collisions: Will quantum computers make SHARCS obsolete?](https://cr.yp.to/hash/collisioncost-20090823.pdf)
+
+## **RSA-OAEP**
+
+### **Analysis**
+
+RSA-OAEP, as any RSA algorithm, **is not quantum resistant**.
+Shor algorithm can be used to solve in a reasonable time 3 mathematical problems on which RSA cryptography is based on : integer factorization problem, the discrete logarithm problem and the elliptic-curve discrete logarithm problem. Therefore, RSA-OAEP is easily breakable for an attacker with quantum capability.
+
+### **Recommendations**
+
+Replace RSA. This recommendation must be the highest priority regarding the post-quantum security of Sealed Secrets.
+There are three serious candidates to use instead of RSA : LMS and XMSS, which are Lattice-based, and McEliece with random Goppa codes, which is code-based and relies on SDP (Syndrome Decoding Problem).
```

Comment:
```suggestion
There are three serious candidates to use instead of RSA: LMS and XMSS, which are Lattice-based, and McEliece with random Goppa codes, which is code-based and relies on SDP (Syndrome Decoding Problem).
```